frontend/src/metabase/nav/containers/MainNavbar/MainNavbar.tsx

```diff
@@ -122,21 +110,7 @@ function MainNavbar({
   );
 }
 
-function maybeGetQuestionId(
-  state: State,
-  { location, params }: MainNavbarOwnProps,
-) {
-  const { pathname } = location;
-  const canFetchQuestion = isQuestionPath(pathname) || isModelPath(pathname);
-  return canFetchQuestion ? Urls.extractEntityId(params.slug) : null;
-}
-
 // eslint-disable-next-line import/no-default-export -- deprecated usage
-export default _.compose(
-  connect(mapStateToProps, mapDispatchToProps),
-  Questions.load({
```

Comment:
This should have worked - if the question was properly cached by the entity framework. Could you investigate why it didn't work?